### ISSUE TITLE
Cypress 12.7.0: Very slow running tests in Azure Devops CI and consistently timing out tests

### ISSUE THREAD

**Issue by Jerry1811** [cypress-io/cypress#26063]
### Current behavior

A successful test run on a local machine ~ 10 mins
Test run on Azure Devops pipeline ~ Test does not complete running and job fails after 60 mins

Browser: electron 106 (headless), firefox 109 (headless)

### Desired behavior

I expect the test to finish running within the 10 mins timeframe

### Test code to reproduce

Not relevant in this case

### Cypress Version

12.7.0

### Node version

v16.16.0

### Operating System

windows

**Comment by RaceWalker** [comment]
Seems like it's linked to this -> https://github.com/cypress-io/cypress/issues/26062

**Comment by nagash77** [comment]
Hello @Jerry1811 , I see in the reproduction steps you said `Not relevant in this case`.  Unfortunately with the information you have provided we will not be able to debug or assist in this issue.  I am closing this until a reproducible example can be provided.

**Comment by lindakumah** [comment]
@nagash77 this behaviour is happening when running the tests in CI. I don't see which part of my test code I should add to this issue

**Comment by Jerry1811** [comment]
@nagash77 This issue is happening when running the tests in the CI and I don't think it has anything to do with the test code since the tests run fine on my local desktop. Reason when I added that comment

**Comment by nagash77** [comment]
Hi @Jerry1811 ,  Thanks for that clarification. The reason we need a reproducible example is because we are not seeing the same thing on our end.  Without a reproducible example that can demonstrate the issue we will not be able to investigate further.  Issues with CI are incredibly complex to troubleshoot and we need to assistance if we are going to spend time digging into a particular issue that is not being widely reported by others.

I am going to close this issue until we have a reproduction.  Feel free to comment here with a reproduction and we can reopen.

**Comment by dlatikaynen** [comment]
Steps to reproduce:
(1) write a big, commercial software product with an Angular frontend
(2) despair
(3) write ~ 100 cypress test specs to cover everything
(4) npx cypress run, local, 10-15 minutes until finishes
(5) npx cypress run, in an Azure Devops pipeline on Ubuntu, 55 minutes until it finishes

**update 2023-12-08** updated to cypress 13.6 today. same codebase, same pipeline, same set of tests  
locally on the same server where the CI build agent runs: **09:40 min**
in the Azure Devops build pipeline on self-hosted Ubuntu: **53:58 min** (counting only the `cypress run` stage, nothing else running in parallel)  
this is repeatable, and persistent.

**Comment by sunshineo** [comment]
Facing similar issue. And it's unclear why this is closed again on Sept 5th

**Comment by jennifer-shehane** [comment]
We've made several performance improvements in v13 of Cypress and are continuing to make more. Please try upgrading. If you're still seeing performance issues - open a new issue with a description, reproducible, and/or a link to a Cloud Test Replay and we can investigate. We are looking into these things!

**Comment by ritikb01** [comment]
I am also facing the same issue. I have 150 cases which takes 2 hours to run locally but in Azure Build pipeline it is taking 5 hours. I have put a timeout of 300 minutes so mostly the build gets cancelled. I am cleaning the build before run as well.
